### ISSUE TITLE
BUG: Issues when Private Practitioner changes jobs and becomes an IRS Respondent

### ISSUE THREAD

**Issue by ttlenard** [ustaxcourt/ef-cms#8042]
**Describe the Bug**

**Apologies...this is kind of a long description for this!**

This user: 66692668-2c07-442c-a10c-4575377aa7b7 Jeffrey Phillips (PJ0955) switched from a private practitioner to an IRS attorney.  We can see that this recently happened (August 9th maybe? - Mike saw this in the DB).  This user updated their own contact info, including their email address.

He had a Tax court BAR number already, and it looks as if Admissions(???) probably edited his employer from Private to IRS.

We think that maybe there was some sort of glitch in this process?  We can query this user and see that his role says IRS Practitioner, but his section still says PrivatePractitioner (see Screen grab below).  

We think that because of this, he has partial IRS capabilities + partial private practitioner capabilities. Mike logged in as him as the IRS practitioner, and when he was entering in to a case, his dropdown options were those of a private practitioner, not an IRS attorney (even though he came in with his IRS account).  IRS respondent typically have two dropdown options for document type, but he  just had an option to file an entry of appearance as a private practitioner representing the petitioners, and it ended up putting him in as petitioners counsel - not respondent counsel.

For Docket records: 13134-21 and 6384-20 - They noticed that an IRS Respondent was in the Petitioners Counsel area, and the Notice of appearance was like what would come in as a private practitioner. When Docket attempted to add him in as the respondent counsel - it would allow her to add him in, but it doesn't "stick".  It will not save to the record.

Also, when Docket attempts to edit the petitioners counsel, she can click remove counsel, and DAWSON shows confirmation that he was removed, but he doesn't actually get removed. His name remains in the petitioners counsel section. (see screen grab below).

When I log in on MIG and TEST as an Admissions clerk, and I attempt to switch an existing private practitioner over to have an employer as IRS, I get the "spinner of death". Mike was looking at the logs in Kibana and it gave an error "user not found".

We would like to know if this happened to any other Private Practitioners that moved over to the IRS Attorney - can we query for this to see if this guy was the only user that this happened to?  If not, we will likely run into this issue again?

Need to clean up this data so that Jeffrey Phillips is no longer a private Practitioner, and need to ensure that the process of a private practitioner switching to IRS is supported.

**Business Impact/Reason for Severity**
High Severity

**In which environment did you see this bug?**

- PROD
- TEST
- MIG

**Who were you logged in as?**

- IRS Practitioner (He actually submitted a helpdesk ticket related to him having different look and feel than his colleagues)
- Docket Clerk
- Admissions Clerk

**What were you doing when you discovered this bug? (Using the application, demoing, smoke tests, testing other functionality, etc.)**
Using the application

**To Reproduce**
Steps to reproduce the behavior - This is how Docket reported things to me. I watched her do this on PROD.
1. Log in as a docket clerk and go to docket record: 13134-21 and 6384-20
2. Click on Docket record. Find the notice of appearance documents - these were filed by respondent, but they came in as it would for a private practitioner. (they edited the record so that it no longer included petitioner info)
3. Next go to case information tab
4. Click on parties tab
5. Notice that Jeffrey Philips is listed as petitioners counsel, not respondent counsel. 
6. Click Edit on the counsel so that you can remove Jeffrey Phillips
7. Click remove counsel.
8. You will get a confirmation that Counsel has been removed, but he doesn't actually get removed.
9. Next, click on respondent counsel in the parties tab.
10. Click add counsel
11. Look up Jeffrey Phillips by his BAR number (PJ0955)
12. Add him
13. All seems to be well, but when you go back into the case, Jeffrey has not been added.

Next as an admissions clerk:
1. go to the Practitioner tab under the advanced search.
2. Type in a last name like "Smith"
3. Select an active private practitioner
4. Change the employer from Private to IRS
5. Notice that you get the spinner of death!

**Expected Behavior**

**Actual Behavior**
A clear and concise description of what actually happened.

- This specific IRS practitioner has different views in DAWSON from other IRS practitioners
- This specific IRS Practitioner seems to have an incorrect section in assigned to him on the DB side?
- Docket is unable to remove this IRS Practitioner from Petitioners Counsel area
- Docket is unable to add this IRS Practitioner to the Respondent Counsel (they can, it just doesn't save)

**Screenshots**

Notice that it says role = IRS Practitioner, but the Section says Private Practitioner.
![image.png](https://images.zenhubusercontent.com/60f99b8ac0d8d04c8aed1aa3/a5f9ccf4-b188-49b3-b4e7-ccc021940c3b)

Notice that it says that petitioners counsel has been removed, but it doesn't go anywhere. It is stuck there:
![image.png](https://images.zenhubusercontent.com/60f99b8ac0d8d04c8aed1aa3/3886fbe2-4740-473c-a750-9bce944f2b92)

**Desktop (please complete the following information):**
 - Browser [e.g. chrome, safari] Edge

**Smartphone (please complete the following information):**

**Cause of Bug, If Known**

**Process for Logging a Bug:**
* Complete the above information
* Add a severity tag (Critical, High Severity, Medium Severity or Low Severity). See below for priority definition. 

**Severity Definition:**
* Critical Defect
Blocks entire system's or module’s functionality
No workarounds available
Testing cannot proceed further without bug being fixed.

* High-severity Defect
Affects key functionality of an application
There's a workaround, but not obvious or easy
App behaves in a way that is strongly different from the one stated in the requirements

* Medium-severity Defect
A minor function does not behave in a way stated in the requirements.
Workaround is available and easy

* Low-severity Defect
Mostly related to an application’s UI
Doesn't need a workaround, because it doesn't impact functionality

## Definition of Done (Updated 4-14-21)
**Product Owner**
 - [ ]  Bug fix has been validated in the Court's test environment

**Engineering**
 - [ ] Automated test scripts have been written
 - [ ] Field level and page level validation errors (front-end and server-side) integrated and functioning
 - [ ] Verify that language for docket record for internal users and external users is identical
 - [ ] New screens have been added to pa11y scripts
 - [ ] All new functionality verified to work with keyboard and macOS voiceover https://www.apple.com/voiceover/info/guide/_1124.html 
 - [ ] READMEs, other appropriate docs, JSDocs and swagger/APIs fully updated
 - [ ] UI should be touch optimized and responsive for external only (functions on supported mobile devices and optimized for screen sizes as required)
 - [ ] Interactors should validate entities before calling persistence methods
 - [ ] Code refactored for clarity and to remove any known technical debt
 - [ ] Deployed to the Court's test environment

**Comment by matthopson** [comment]
It looks like we had a story related to this that was closed without doing any work on it.
https://github.com/flexion/ef-cms/issues/7635

Do we know for sure this is a bug and not something that we hadn't actually performed work on? I'll keep looking to see if there are any other similar stories that may have touched on this.

**Comment by ttlenard** [comment]
After a deep dive, we have determined that Mr. Phillips filed the EA while his profile had him under Private as his employer.  His employment did get switched to IRS, but this was after he had filed those petitions.  This caused some issues where Docket or Admissions could not remove him from private counsel.  Solution: Admissions will switch his employer back to private, then they will be able to remove him from private counsel in the "bad cases". Then admissions will switch his employer back to IRS.  This should solve the problem!

I submitted an associated story so that this will not be able to happen again. It is 8885
